File: R/S3-3.R
# Prueba de proporciones para comparar con un valor teórico

# Datos del problema (ejemplo: eficacia de un medicamento)
n <- 100       # Total de casos
x <- 70        # Casos exitosos
p0 <- 0.60     # Proporción teórica de referencia
alpha <- 0.05  # Nivel de significancia

# Realizar prueba Z de proporción (unilateral derecha)
prop_test_result <- prop.test(x = x, 
                              n = n, 
                              p = p0, 
                              alternative = "greater", 
                              correct = FALSE)

# Extraer resultados
test_stat <- prop_test_result$statistic  # Estadístico Z
p_value <- prop_test_result$p.value      # Valor p

# Resultados detallados
cat("----- Resultados de la Prueba de Proporción -----\n")
cat(sprintf("Proporción observada: %.2f\n", x/n))
cat(sprintf("Proporción teórica: %.2f\n", p0))
cat(sprintf("Estadístico Z: %.4f\n", test_stat))
cat(sprintf("Valor p: %.6f\n", p_value))

# Decisión estadística
cat("\n----- Conclusión -----\n")
if (p_value < alpha) {
  cat("Rechazamos H₀ (p =", format.pval(p_value), "< α =", alpha, ")\n")
  cat("Conclusión: El nuevo medicamento es significativamente mejor.\n")
} else {
  cat("No rechazamos H₀ (p =", format.pval(p_value), "≥ α =", alpha, ")\n")
  cat("Conclusión: No hay evidencia suficiente para afirmar que el nuevo medicamento es mejor.\n")
}

# Opcional: Intervalo de confianza
conf_int <- prop.test(x, n, conf.level = 0.95)$conf.int
cat(sprintf("\nIntervalo de confianza 95%%: [%.4f, %.4f]", conf_int[1], conf_int[2]))

File: Python/S3/S3-3.py
#Un investigador de mercado está realizando un estudio encargado por una concesionaria de automóviles que tiene sucursales en todo el país, para conocer acerca de las reparaciones de autos. En particular, tiene interés en averiguar cuál es el gasto promedio de las reparaciones. Para estimar cuántos casos incluir en la muestra, debe considerar un valor de la desviación estándar. Basado en su experiencia cree apropiado considerar una desviación estándar de U$200 del gasto realizado en reparaciones.
#No obstante, decide corroborar su hipótesis a través de una encuesta piloto a 25 dueños de automóviles. Una vez que lleva a cabo este estudio preliminar, resulta una desviación estándar de la muestra de U$237,52.
#¿Aún puede trabajar con su idea previa de la desviación estándar para calcular el tamaño muestral?
#Utilice una significancia del 10%.

import scipy.stats as stats

# Prueba de hipótesis para la varianza poblacional (Python)

# Datos del problema
n = 25                     # Tamaño muestral
sigma_hipotetica = 200     # Desviación estándar hipotética
s_muestral = 237.52        # Desviación estándar muestral
alpha = 0.10               # Nivel de significancia

# Calcular el estadístico Chi-cuadrado
chi2_stat = (n - 1) * (s_muestral ** 2) / (sigma_hipotetica ** 2)

# Calcular el valor crítico Chi-cuadrado (unilateral derecho)
chi2_critical = stats.chi2.ppf(1 - alpha, df=n - 1)

# Resultados de la prueba
print("----- Prueba de Hipótesis para la Varianza -----")
print(f"Tamaño muestral (n): {n}")
print(f"Desviación estándar muestral: {s_muestral:.2f}")
print(f"Desviación estándar hipotética: {sigma_hipotetica:.2f}")
print(f"\nEstadístico Chi-cuadrado: {chi2_stat:.4f}")
print(f"Valor crítico (α={alpha}): {chi2_critical:.4f}")

# Calcular valor p
p_value = 1 - stats.chi2.cdf(chi2_stat, df=n - 1)
print(f"Valor p: {p_value:.6f}")

# Decisión estadística
print("\n----- Conclusión -----")
if chi2_stat > chi2_critical:
    print(f"Rechazamos H₀ (χ² = {chi2_stat:.2f} > {chi2_critical:.2f})")
    print("Conclusión: La variabilidad real es significativamente mayor que la hipotética")
else:
    print(f"No rechazamos H₀ (χ² = {chi2_stat:.2f} ≤ {chi2_critical:.2f})")
    print("Conclusión: No hay evidencia suficiente contra la variabilidad hipotética")

# Intervalo de confianza para la varianza (opcional)
lower_var = (n - 1) * (s_muestral ** 2) / stats.chi2.ppf(1 - alpha/2, n - 1)
upper_var = (n - 1) * (s_muestral ** 2) / stats.chi2.ppf(alpha/2, n - 1)
print(f"\nIntervalo de confianza {(1-alpha)*100}% para la varianza:")
print(f"[{lower_var:.2f}, {upper_var:.2f}]")

